extra/material.js

```diff
@@ -2457,12 +2457,6 @@ function MaterialTab(tab, ctx) {
         }
         tab.addEventListener('click', function (e) {
             e.preventDefault();
-            var href = tab.href.split('#')[1];
-            var panel = ctx.element_.querySelector('#' + href);
-            ctx.resetTabState_();
-            ctx.resetPanelState_();
-            tab.classList.add(ctx.CssClasses_.ACTIVE_CLASS);
-            panel.classList.add(ctx.CssClasses_.ACTIVE_CLASS);
```

Comment:
tbh, I think the best thing to do here is submit a PR in mdl to add a check before reseting everything. (it would be a safe check for them but 
And then we should be able to upgrade the element :)